Packs/CommonScripts/Scripts/CreateNewIndicatorsOnly/CreateNewIndicatorsOnly.py

```diff
@@ -9,22 +10,41 @@
 
 KEY_CREATION_STATUS = 'CreationStatus'
 
+MAX_FIND_INDICATOR_RETRIES = 10
+SLEEP_TIME = 2
+
 
 def associate_indicator_to_incident(indicator_value: Any) -> None:
     """
     Associate an indicator to this incident. Raise an exception if an error occurs.
     """
 
     incident_id = demisto.incidents()[0].get('id')
+    demisto.debug(f"The incident id is {incident_id}")
 
     cmd_args = {
         'incidentId': incident_id,
         'value': f"{indicator_value}"  # Force an error
     }
 
-    res = execute_command('associateIndicatorToIncident', cmd_args)
-
-    if (res != 'done'):
+    retry_num = 1
+    res = ''
+    while res != 'done' and retry_num <= MAX_FIND_INDICATOR_RETRIES:
+        try:
+            demisto.debug(f"Executing associateIndicatorToIncident command with retry {retry_num}/{MAX_FIND_INDICATOR_RETRIES}")
+            res = execute_command('associateIndicatorToIncident', cmd_args)
+            demisto.debug(f"The associateIndicatorToIncident response is: {res}")
+
+        except Exception as err:
+            if "For associateIndicatorToIncident found no indicator" not in str(err):
+                raise err
+            # else log and continue with retries after sleeping
+            demisto.debug(f"Failed to find indicator {indicator_value} in the system.")
+            if retry_num != MAX_FIND_INDICATOR_RETRIES:
+                time.sleep(SLEEP_TIME)  # pylint: disable=E9003
+            retry_num += 1
+
+    if res != 'done':
```

Comment:
execute command should return an entry no? I see that your code looks for a string

Reply:
you're thinking of `demisto.executeCommand`
`execute_command` on the other hand fails on error and returns the value of the command (in this case the string 'done' on success)

Reply:
nice